client/src/main/java/io/pravega/client/netty/impl/ConnectionFactory.java

```diff
@@ -30,6 +31,15 @@
      */
     CompletableFuture<ClientConnection> establishConnection(PravegaNodeUri endpoint, ReplyProcessor rp);
```

Comment:
In which case are we still going to need this? I'm not necessarily suggesting we remove it or deprecate it, I'm only trying to understand.

Reply:
The SegmentHelper on the controller using this api.